sql/core/src/test/scala/org/apache/spark/sql/UnwrapCastInComparisonEndToEndSuite.scala

```diff
@@ -191,18 +191,27 @@ class UnwrapCastInComparisonEndToEndSuite extends QueryTest with SharedSparkSess
   }
 
   test("SPARK-36607: Support BooleanType in UnwrapCastInBinaryComparison") {
+    def verifyCastBoolean(df: DataFrame, query: String, answer: Seq[Row]): Unit =
+      if (conf.ansiEnabled) {
+        intercept[AnalysisException] {
+          df.where(query).collect()
```

Comment:
It looks meaningless to ansi error behavior in this optimizer rule test suite. Can we simply turn off ansi in this test case?

Reply:
updated